axum-macros/src/debug_handler.rs

```diff
@@ -159,6 +159,12 @@ fn check_inputs_impls_from_request(
     body_ty: &Type,
     state_ty: Type,
 ) -> TokenStream {
+    let takes_self = item_fn
+        .sig
+        .inputs
+        .first()
+        .map_or(false, |arg| matches!(arg, FnArg::Receiver(_)));
```

Comment:
Should also check for the other `FnArg` variant where the `self` keyword is followed by a colon and type like any other parameter.

Reply:
Ah yes good catch!